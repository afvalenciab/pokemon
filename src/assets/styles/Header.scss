
.header {
  display: flex;
  align-items: center;

  h1 {
    margin: 0; 
  }

  figure {
    margin: 0;
    margin-right: 10px;
    img {
      vertical-align: top;
      width: 50px;
    }
  }
}