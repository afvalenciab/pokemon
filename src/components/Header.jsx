import React from 'react';
import logo from '../assets/static/Pookeball_logo.svg';
import '../assets/styles/Header.scss';

const Header = () => {
  return (
    <section className='header'>
      <figure>
        <img src={logo} alt='Pokemon Logo' />
      </figure>
      <h1>Pok√©mon</h1>
    </section>
  );
};

export default Header;
