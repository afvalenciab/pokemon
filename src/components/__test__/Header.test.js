import React from 'react';
import { create } from 'react-test-renderer';
import { shallow, configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';

configure({ adapter: new Adapter() });

describe('Header Testing', () => {
  test('Match Snapshot', () => {
    const header = create(<Header />);
    expect(header.toJSON()).toMatchSnapshot();
  });

  test('Header has class .header', () => {
    const header = shallow(<Header />);
    const headerElm = header.find('.header');
    expect(headerElm.hasClass('header')).toBe(true);
  });

  test('Header has 1 figure tag', () => {
    const header = render(<Header />);
    expect(header.find('figure')).toHaveLength(1);
  });
});
