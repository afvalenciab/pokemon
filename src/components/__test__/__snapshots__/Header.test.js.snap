// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Testing Match Snapshot 1`] = `
<section
  className="header"
>
  <figure>
    <img
      alt="Pokemon Logo"
      src="test-file-stubs"
    />
  </figure>
  <h1>
    Pok√©mon
  </h1>
</section>
`;
